### PR TITLE
Update src/library/Button/index.d

### PR DESCRIPTION
**Pull request by tx-steven**

### CODE REVIEW

File: src/library/Button/index.d.ts [hunk 1]
```diff
@@ -1 +1,36 @@
...
+import {
+  ElementType,
+  ReactElement,
+  ReactNode,
+  Ref,
+  SyntheticEvent
+} from 'react';
+
+export declare namespace Button {
+  type Size = 'small' | 'medium' | 'large' | 'jumbo';
```
Comment (tx-nestor) [inline src/library/Button/index.d.ts]: if they don't need to be exported then you can move out of namespace and define locally only.

File: src/library/Button/index.d.ts [hunk 2]
```diff
@@ -1 +1,36 @@
...
+  type Variant = 'danger' | 'grayscale' | 'success' | 'warning';
+
+  export interface Props {
+    as?: ElementType;
+    children?: ReactNode;
+    circular?: boolean;
+    css?: { [key: string]: any };
+    disabled?: boolean;
+    forwardedRef?: Ref<any>;
```
Comment (tx-nestor) [inline src/library/Button/index.d.ts]: are we using this?

File: src/library/Button/index.d.ts [hunk 3]
```diff
@@ -1 +1,36 @@
...
+    onClick?: (event: SyntheticEvent) => void;
+    primary?: boolean;
+    size?: Size;
+    type?: string;
+    variant?: Variant;
+    width?: string;
+  }
+}
+
+export declare function Button(props: Button.Props): ReactElement;
```
Comment (tx-nestor) [inline src/library/Button/index.d.ts:28]: This doesn't match what @tx-trent did for `Icon`.
```
export declare const Icon: (props: Icon.Props) => React.ReactElement<any, string | ((props: any) => React.ReactElement<any, string | any | (new (props: any) => React.Component<any, any, any>)> | null) | (new (props: any) => React.Component<any, any, any>)>;
```

Not sure if we need all of that, but let's be consistent and use the same approach.

**Review by tx-nestor — APPROVED** [review]
Approved with suggestions/questions

File: src/library/Button/index.d.ts [hunk 4]
```diff
@@ -1 +1,29 @@
...
+    circular?: boolean;
+    css?: { [key: string]: any };
+    disabled?: boolean;
+    fullWidth?: boolean;
+    href?: string;
+    iconEnd?: ReactElement;
+    iconStart?: ReactElement;
+    justifyContent?: string;
+    minimal?: boolean;
+    onClick?: (event: SyntheticEvent) => void;
```
Comment (tx-michael) [inline src/library/Button/index.d.ts:19]: @tx-nestor @tx-steven 
This change on 19 is breaking in multiple projects. TX-Print breaks in several components and If I install this in `td-core-web-ui`, that project also breaks. I don't see any open pr's to address this in core-web-ui. How would we refactor the projects to conform to this type? I'm unable to adapt my usage of event in tx-print to this type on line 19.

Comment (tx-michael) [reply src/library/Button/index.d.ts:19]: In td-core-web-ui

image

Comment (tx-michael) [reply src/library/Button/index.d.ts:19]: In tx-print:
image


If I change `HTMLButtonElement` to `Eleement`, I then lose access to the `value` property on event.target

**Pull request merged**